application/controllers/EventRuleController.php

```diff
@@ -286,17 +299,85 @@ public static function createFilterString(Filter\Rule $filters): string
         return ! empty($filterStr) ? $filterStr : '';
     }
 
+    public function addAction(): void
+    {
+        $this->addTitleTab(t('Add Event Rule'));
+        $eventRule = $this->params->toArray(false);
+        $this->getTabs()->setRefreshUrl(Url::fromRequest());
+        $this->addContent(Html::tag('div', ['class' => 'container', 'id' => 'dummy']));
+
+        $this->controls->addAttributes(['class' => 'event-rule-detail']);
+        $ruleId = $eventRule['id'] ?? '-1';
+
+        if ($this->getRequest()->isPost()) {
+            if ($this->getRequest()->has('searchbar')) {
+                $eventRule['object_filter'] = $this->getRequest()->get('searchbar');
+            } else {
+                $eventRule['object_filter'] = '';
+            }
+        }
+
+        $eventRuleConfigSubmitButton = (new SubmitButtonElement(
+            'save',
+            [
+                'label'          => t('Add Event Rule'),
+                'form'           => 'event-rule-config-form'
+            ]
+        ))->setWrapper(new HtmlElement('div', Attributes::create(['class' => ['icinga-controls', 'save-config']])));
+
+        $eventRuleConfig = new EventRuleConfigForm(
+            $eventRule,
+            Url::fromPath(
+                'notifications/event-rule/search-editor',
+                ['id' => $ruleId, 'object_filter' => $eventRule['object_filter'] ?? '']
+            )
+        );
+
+        $eventRuleConfig
+            ->on(Form::ON_SUCCESS, function (EventRuleConfigForm $form) use ($eventRule) {
+                /** @var string $ruleId */
+                $ruleId = $eventRule['id'];
+                /** @var string $ruleName */
+                $ruleName = $eventRule['name'];
+                $eventRuleConfig = array_merge($eventRule, $form->getValues());
+                $form->addOrUpdateRule($ruleId, $eventRuleConfig);
+                Notification::success(sprintf(t('Successfully add event rule %s'), $ruleName));
+                $this->redirectNow('__CLOSE__');
+            })
+            ->handleRequest($this->getServerRequest());
+
+        $eventRuleForm = Html::tag('div', ['class' => 'event-rule-form'], [
+            Html::tag('h2', $eventRule['name'] ?? ''),
+            Html::tag(
+                'div',
+                [
+                    'class' => 'not-allowed',
+                    'title' => $this->translate('Cannot edit event rule until it is saved to database')
+                ],
+                (new Link(
+                    new Icon('edit'),
+                    Url::fromPath('notifications/event-rule/edit', [
+                        'id' => -1
+                    ]),
+                    ['class' => ['control-button', 'disabled']]
+                ))->openInModal()
+            )
```

Comment:
Makes sense.

Reply:
As explained offline, since not much in the detail view as to be changed once the event rule is added. We will retain the edit link, but will just toggle between disabled and not disabled.